q2_longitudinal/_longitudinal.py

```diff
@@ -508,3 +506,62 @@ def first_distances(distance_matrix: skbio.DistanceMatrix,
         metadata, state_column, individual_id_column, metric=None,
         replicate_handling=replicate_handling, baseline=baseline,
         distance_matrix=distance_matrix)
+
+
+def feature_volatility(ctx,
+                       table,
+                       metadata,
+                       state_column,
+                       individual_id_column=None,
+                       cv=5,
+                       random_state=None,
+                       n_jobs=1,
+                       n_estimators=100,
+                       estimator='RandomForestRegressor',
+                       parameter_tuning=False,
+                       missing_samples='error'):
+    regress = ctx.get_action('sample_classifier', 'regress_samples')
+    # TODO: Add this back once filter_features can operate on a
+    # FeatureTable[RelativeFrequency] artifact (see notes below)
+    # filter_tab = ctx.get_action('feature_table', 'filter_features')
+    relative = ctx.get_action('feature_table', 'relative_frequency')
+    volatility = ctx.get_action('longitudinal', 'plot_feature_volatility')
+
+    # this validation must be tested here ahead of supervised regression
+    states = metadata.get_column(state_column)
+    if not isinstance(states, qiime2.NumericMetadataColumn):
+        raise TypeError('state_column must be numeric.')
+
+    estimator, importances, predictions, summary, accuracy = regress(
+        table, metadata=states, cv=cv, random_state=random_state,
+        n_jobs=n_jobs, n_estimators=n_estimators, estimator=estimator,
+        parameter_tuning=parameter_tuning, optimize_feature_selection=True,
+        missing_samples=missing_samples)
+
+    # filter table to important features and convert to relative frequency
+    feature_md = importances.view(qiime2.Metadata)
+    filtered_table, = relative(table=table)
+    # TODO: use feature_table.relative_frequency to convert to transform
+    # once feature_table.filter_features can accept a relative frequency table.
+    # We must transform and then filter, because otherwise relative frequencies
+    # are based only on filtered features, which can seriously distort
+    # frequency if a large number of features is filtered out. At which point
+    # we can just filter with this code:
+    # filtered_table, = filter_tab(table=filtered_table, metadata=feature_md)
+    # filter_features also seems problematic, since it drops SAMPLES that have
+    # none of the features to keep... distorting averages in the control chart.
+    # For now let's use biom for this:
+    filtered_table = filtered_table.view(biom.Table).filter(
+        ids_to_keep=feature_md.get_ids(), axis='observation', inplace=False)
+    filtered_table = qiime2.Artifact.import_data(
```

Comment:
This should be `ctx.make_artifact` instead, so that garbage collection on error works correctly.